linux_os/guide/system/apparmor/all_apparmor_profiles_enforced/oval/shared.xml

```diff
@@ -0,0 +1,53 @@
+<def-group>
+  <definition class="compliance" id="{{{ rule_id }}}" version="1">
+      {{{ oval_metadata("Enforce all AppArmor Profiles") }}}
+    <criteria comment="Enforce all AppArmor Profiles" operator="AND">
+      <criterion comment="Apparmor profiles are loaded" test_ref="test_profiles_are_loaded" />
+      <criterion comment="No profiles are in complain mode." test_ref="test_profiles_are_in_enforce_mode" />
+      <criterion comment="No processes which are unconfined." test_ref="test_processes_are_confined" />
+    </criteria>      
+  </definition>
+
+    <!-- Define 1st test for all apparmor profiles. -->
+  <linux:apparmorstatus_test id="test_profiles_are_loaded" version="1"
+    check="all" check_existence="at_least_one_exists"
+    comment="Verify that profiles are loaded" >
+    <linux:object object_ref="object_apparmor_profiles_existing" /> 
+    <linux:state state_ref="apparmor_profiles_are_loaded" />
+  </linux:apparmorstatus_test>
+
+  <!-- Define 2nd test for all apparmor profiles. -->
+  <linux:apparmorstatus_test id="test_profiles_are_in_enforce_mode" version="1"
+    check="all" check_existence="at_least_one_exists"
+    comment="Zero profiles are in complain mode" >
+    <linux:object object_ref="object_apparmor_profiles_existing" /> 
+    <linux:state state_ref="no_apparmor_profiles_in_complian_mode" />
+  </linux:apparmorstatus_test>
+
+  <!-- Define "3rd" test for processes. -->
+  <linux:apparmorstatus_test id="test_processes_are_confined" version="1"
+    check="all" check_existence="at_least_one_exists"
+    comment="Zero processes are unconfined" >
+    <linux:object object_ref="object_apparmor_profiles_existing" /> 
+    <linux:state state_ref="no_unconfined_processes" />
+  </linux:apparmorstatus_test>
+
+  <linux:apparmorstatus_object id="object_apparmor_profiles_existing" version="1">
+  </linux:apparmorstatus_object >
```

Comment:
you can consolidate this in one line:
`<linux:apparmorstatus_object id="object_apparmor_profiles_existing" version="1" />`